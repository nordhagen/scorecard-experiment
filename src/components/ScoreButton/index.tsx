import * as React from 'react'
import styles from './style.scss'

interface Props {
  id: string
  color: string
  clickDelegate: Function
}

const ScoreButton: React.FunctionComponent<Props> = props => {
  let inlineStyle = {
    backgroundColor: props.color
  }
  return (
    <button
      className={styles.ScoreButton}
      onClick={e => props.clickDelegate(props.id)}
      style={inlineStyle}
    >
      {props.id}
    </button>
  )
}

export default ScoreButton
