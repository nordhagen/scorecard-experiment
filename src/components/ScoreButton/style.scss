@import '../../global.scss';

.ScoreButton {
  width: 100px;
  height: 100px;
  display: inline-flex;
  margin: 10px;
  flex-direction: column;
  border: 0;
  background: #ccc;
  justify-content: center;
  font-family: sans-serif;
  align-items: center;
  font-size: 2rem;
  cursor: pointer;
}
