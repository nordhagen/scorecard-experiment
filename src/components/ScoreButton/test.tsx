import React from 'react'
import ScoreButton from './index'
import renderer from 'react-test-renderer'

test('It renders correctly', () => {
  const pastItem = renderer.create(
    <ScoreButton id="a" color="#ff0000" clickDelegate={console.log} />
  )

  expect(pastItem.toJSON()).toMatchSnapshot()
})
