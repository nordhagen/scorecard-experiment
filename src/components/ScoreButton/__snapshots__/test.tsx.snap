// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#ff0000",
    }
  }
>
  a
</button>
`;
